### PR TITLE
feature_addition : Added a mean squared loss layer to DNN with test case

### PR DESCRIPTION
**Pull request by dennisfrancis**
Added mean squared loss layer "loss_mean_squared" to
DNN as requested in https://github.com/davisking/dlib/issues/152
Also added test case of a simple linear regression with one variable
that uses this layer.

### CODE REVIEW

**Comment by davisking** [comment]
Awesome, looks good :)

Some parts have inconsistent indenting due to the presence of tab characters.  Can you fix the code so the indenting is uniform?  Then the PR will be ready to merge.

**Comment by dennisfrancis** [comment]
Thanks for pointing out, I have made the change.

File: dlib/dnn/loss.h [hunk 1]
```diff
@@ -1294,6 +1294,113 @@ namespace dlib
...
+                             output_tensor.nc() == 1 &&
+                             output_tensor.k() == 1);
+                DLIB_CASSERT(input_tensor.num_samples() == output_tensor.num_samples());
+
+                const float* out_data = output_tensor.host();
+                for (long i = 0; i < output_tensor.num_samples(); ++i)
+                {
+                    *iter++ = out_data[i];
+                }
+            }
```
Comment (davisking) [inline dlib/dnn/loss.h]: Tabbing is still weird :)

Comment (dennisfrancis) [inline dlib/dnn/loss.h]: hmm, looks like my emacs settings indents templates in a particular way. I have manually adapted to dlib style. Hopefully I fixed all of them :)

**Comment by davisking** [comment]
Sweet, looks good.  Thanks for the PR :)

**Pull request merged**